src/with-component.js

```diff
@@ -2,9 +2,10 @@ import { render } from 'preact';
 import { HTMLElement } from './util';
 import { withProps } from './with-props';
 import { withRender } from './with-render';
+import { withUnique } from './with-unique';
 
 export function withComponent (Base = HTMLElement) {
-  return class extends withRender(withProps(Base)) {
+  return class extends withUnique(withRender(withProps(Base))) {
```

Comment:
hmm well I guess mixins are our only option if we don't wanna directly mutate user defined component class.

From statically typed js point of view -> using decorators wont add props and other methods definition to class type definition